ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -0,0 +1,376 @@
+        csv_pre_upgrade = CSV(
+            resource_name=csv_name_pre_upgrade,
+            namespace=self.namespace
+        )
+        return get_images(csv_pre_upgrade.get())
+
+    def set_upgrade_channel(self):
+        """
+        Wait for the new package manifest for upgrade.
+
+        Returns:
+            str: OCS subscription channel
+
+        """
+        operator_selector = get_selector_for_ocs_operator()
+        package_manifest = PackageManifest(
+            resource_name=OCS_OPERATOR_NAME, selector=operator_selector,
+        )
+        package_manifest.wait_for_resource()
+        channel = config.DEPLOYMENT.get('ocs_csv_channel')
+        if not channel:
+            channel = package_manifest.get_default_channel()
+
+        return channel
+
+    def update_subscription(self, channel):
+        """
+        Updating OCS operator subscription
+
+        Args:
+            channel: (str): OCS subscription channel
+
+        """
+        subscription = OCP(
+            resource_name=constants.OCS_SUBSCRIPTION,
+            kind='subscription',
+            namespace=config.ENV_DATA['cluster_namespace'],
+        )
+        current_ocs_source = subscription.data['spec']['source']
+        log.info(
+            f"Current OCS subscription source: {current_ocs_source}"
+        )
+        ocs_source = current_ocs_source if self.upgrade_in_current_source else (
+            constants.OPERATOR_CATALOG_SOURCE_NAME
+        )
+        patch_subscription_cmd = (
+            f'patch subscription {constants.OCS_SUBSCRIPTION} '
+            f'-n {self.namespace} --type merge -p \'{{"spec":{{"channel": '
+            f'"{channel}", "source": "{ocs_source}"}}}}\''
+        )
+        subscription.exec_oc_cmd(patch_subscription_cmd, out_yaml_format=False)
+
+        subscription_plan_approval = config.DEPLOYMENT.get(
+            'subscription_plan_approval'
+        )
+        if subscription_plan_approval == 'Manual':
+            wait_for_install_plan_and_approve(self.namespace)
+
+    def check_if_upgrade_completed(self, channel, csv_name_pre_upgrade):
+        """
+        Checks if OCS operator finishes it's upgrade
+        Args:
+            channel: (str): OCS subscription channel
+            csv_name_pre_upgrade: (str): OCS operator name
+
+        Returns:
+            bool: True if upgrade completed
+
+        """
+        operator_selector = get_selector_for_ocs_operator()
+        package_manifest = PackageManifest(
+            resource_name=OCS_OPERATOR_NAME, selector=operator_selector,
+        )
+        csv_name_post_upgrade = package_manifest.get_current_csv(channel)
+        if csv_name_post_upgrade == csv_name_pre_upgrade:
+            log.info(f"CSV is still: {csv_name_post_upgrade}")
+            return False
+        else:
+            log.info(f"CSV now upgraded to: {csv_name_post_upgrade}")
+            return True
+
+    def check_if_images_updated(self, channel, pre_upgrade_images, upgrade_version):
```

Comment:
the method name doesn't reflect its return value, which is not bool

Reply:
fixed